demo-circuit/src/naive_threshold_sig_w_key_rotation/tests/mod.rs

```diff
@@ -0,0 +1,1340 @@
+            expected_threshold_results,
+            change_signing_keys,
+            change_master_keys,
+        ) = generate_circuit_test_data(condition);
+        println!("valid_sigs = {}, threshold = {}", valid_sigs, threshold);
+
+        let circuit = get_test_circuit_instance(
+            max_pks,
+            num_custom_fields,
+            valid_sigs,
+            threshold,
+            is_first_certificate,
+            change_signing_keys,
+            change_master_keys,
+        );
+
+        let failing_constraint =
+            test_naive_threshold_circuit(circuit, debug_only, max_pks, num_custom_fields, None);
+        println!("Failing constraint: {:?}", failing_constraint);
+        if expected_threshold_results {
+            assert!(failing_constraint.is_none());
+        } else {
+            assert!(failing_constraint.is_some());
+        }
+    }
+}
+
+#[serial]
+#[test]
+fn test_naive_threshold_sig_w_key_rotation_circuit_debug_only() {
+    let variants = [
+        TestDataType::AllRandomFirstCert,
+        TestDataType::AllRandomNextCert,
+        TestDataType::ValidSigEqThresholdNextCert,
+        TestDataType::ValidSigGreaterThresholdNextCert,
+        TestDataType::ValidSigSmallerThresholdNextCert,
+        TestDataType::ValidSigAndThresholdEqZeroNextCert,
+        TestDataType::ValidSigAndThresholdEqValidatorsNextCert,
+        TestDataType::ValidSigEqValidatorsThresholdEqZeroNextCert,
+        TestDataType::ValidSigEqZeroThresholdEqValidatorsNextCert,
+        TestDataType::ChangeMasterKeysNextCert,
+        TestDataType::ChangeSigningAndMasterKeysNextCert,
+    ];
+
+    for condition in variants.iter().cloned() {
+        println!("Checking variant {:?}", condition);
+
+        let (
+            max_pks,
+            num_custom_fields,
+            valid_sigs,
+            threshold,
+            is_first_certificate,
+            expected_threshold_results,
+            change_signing_keys,
+            change_master_keys,
+        ) = generate_circuit_test_data(condition);
+        println!("valid_sigs = {}, threshold = {}", valid_sigs, threshold);
+
+        let circuit = get_test_circuit_instance(
+            max_pks,
+            num_custom_fields,
+            valid_sigs,
+            threshold,
+            is_first_certificate,
+            change_signing_keys,
+            change_master_keys,
+        );
+
+        let failing_constraint =
+            test_naive_threshold_circuit(circuit, true, max_pks, num_custom_fields, None);
+
+        println!("Failing constraint: {:?}", failing_constraint);
+        if expected_threshold_results {
+            assert!(failing_constraint.is_none());
+        } else {
+            assert!(failing_constraint
+                .unwrap()
+                .eq("threshold check/conditional_equals"));
+        }
+    }
+}
+
+//******Test wrong genesis_constant
+#[serial]
+#[test]
+fn test_naive_threshold_sig_w_key_rotation_circuit_wrong_genesis_constant() {
```

Comment:
I would add to the negative tests also the variant where the wrong inputs are provided for the first certificate, to increase coverage also for the corner case of the first certificate